### PR TITLE
Feature/system message

### PR DESCRIPTION
**Pull request by rostum**

### CODE REVIEW

File: game/game/api/world.py [hunk 1]
```diff
@@ -28,6 +31,13 @@ async def enter_world(request: GameRequest):
 
     request.session.send_packet(game.packets.ItemList(items=character.inventory.items))
 
+    welcome = SystemMessage(type=WELCOME_TO_LINEAGE)
+    WORLD.send_sys_message(character, welcome)
+
+    msg_values = [ZoneName(x=-71338, y=258271, z=-3104)]
+    territory = SystemMessage(type=LOC_TI_S1_S2_S3, values=msg_values)
```
Comment (Yurzs) [inline game/game/api/world.py]: why is this static?

File: game/game/models/structures/system_message.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+@dataclass(kw_only=True)
+class MessageValue(BaseDataclass):
+    type: int
+
+
+@dataclass(kw_only=True)
+class Text(MessageValue):
+    type: int = field(default=0, init=False)
```
Comment (Yurzs) [inline game/game/models/structures/system_message.py]: use `ctype.int32` in models

File: game/game/packets/system_message.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@dataclass(kw_only=True)
+class SystemMessagePacket(GameServerPacket):
+    type: ctype.int8 = field(default=100, init=False)
+    message: SystemMessage
+
+    def encode(self, session):
+        encoded = bytearray(self.type)
+        encoded.extend(bytes(ctype.int32(self.message.type)))
+        encoded.extend(bytes(ctype.int32(len(self.message.values))))
```
Comment (Yurzs) [inline game/game/packets/system_message.py]: please use `extend_bytearray` function to extend bytearray

File: game/game/packets/system_message.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        for value in self.message.values:
+            encoded.extend(bytes(ctype.int32(value.type)))
+
+            if isinstance(value, ZoneName):
+                extend_bytearray(
+                    encoded,
+                    [
+                        ctype.int32(value.x),
+                        ctype.int32(value.y),
+                        ctype.int32(value.z)
```
Comment (Yurzs) [inline game/game/packets/system_message.py]: no need to convert when model will have `ctype.int32` for field types

**Review by Yurzs — CHANGES_REQUESTED** [review]

File: game/game/api/world.py (hunk 1 above)
Comment (rostum) [reply game/game/api/world.py]: sorry, it was for testing purpose, forgot to remove.

Comment (rostum) [reply game/game/api/world.py]: it was removed in [this commit](https://github.com/Yurzs/L2py/pull/23/commits/ca82c2bab8cbc72cb5d33d3bc11de984b2b6dd27)

File: game/game/packets/system_message.py (hunk 3 above)
Comment (rostum) [reply game/game/packets/system_message.py]: it was done in [this commit](https://github.com/Yurzs/L2py/pull/23/commits/ca82c2bab8cbc72cb5d33d3bc11de984b2b6dd27)

**Review by Yurzs — APPROVED** [review]

**Comment by Yurzs** [comment]
rebase please

**Pull request merged**